client/v2/client.go

```diff
@@ -0,0 +1,371 @@
+package client
+
+import (
+	"bytes"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"sync"
+	"time"
+
+	"github.com/influxdb/influxdb/models"
+)
+
+type Config struct {
+	// URL of the InfluxDB database
+	URL *url.URL
+
+	// Username is the influxdb username, optional
+	Username string
+
+	// Password is the influxdb password, optional
+	Password string
+
+	// UserAgent is the http User Agent, defaults to "InfluxDBClient"
+	UserAgent string
```

Comment:
User agent should really be the program they wrote, which if you had several programs that used the client library tracking the offender down in the logs would be difficult (if not impossible).  This is just good etiquette letting the consumer of this library have tighter control over what the server will put in the logs as a user agent when it receives requests.